DiscountDiscover/View Controllers/MapViewController.m

```diff
@@ -39,4 +42,22 @@ - (void)addDealMarkers {
     }
 }
 
+- (void)mapView:(GMSMapView *)mapView didTapInfoWindowOfMarker:(GMSMarker *)marker {
+    
+    NSUInteger i = [self.deals indexOfObjectPassingTest:^BOOL(Deal *deal, NSUInteger idx, BOOL *stop) {
+        return [deal.name isEqual:marker.snippet];
```

Comment:
You shouldn't compare objects by name/description, use an ID instead. You can also compare objects directly by memory address if possible to make sure they're exactly the same object. Adding a `DealMarker` class as I suggested before would also help here by giving you direct access to the deal from the marker. You should also check if you can attach a random object to the marker besides the snippet and title, if you could do something like `marker.object = DealMarkerInstance` then you have direct access to the deal without iterating though an array.

Reply:
See the [`userData`](https://developers.google.com/maps/documentation/ios-sdk/reference/interface_g_m_s_overlay#a900ff2be989386d501ed6c64ea1d7ab3) property, you can set that to the `Deal` instance that the marker represents and avoid the iteration and comparing deals.